packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,82 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
 use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
+    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result, Error},
     proto::ExperimentId,
 };
 
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
+    Success,
+    Error,
+}
+
 // TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
-pub fn cleanup_experiment(experiment_id: &ExperimentId) -> Result<()> {
+pub fn cleanup_experiment(
+    experiment_id: &ExperimentId,
+    exit_status: EngineExitStatus,
+) -> Result<()> {
     MemoryId::clean_up(experiment_id)?;
```

Comment:
I don't think we should return an error if the `glob` failed. It's probably the best to modify `Memoryid::clean_up` directly to not return an error at all.

Reply:
Stopped returning https://github.com/hashintel/hash/pull/477/commits/46593ef548a477ffb61a3e8452d3d9c683ba4e86